Loader/index.jsx

```diff
@@ -12,18 +12,72 @@ const renderText = (children) => {
   }
 };
 
-const Loader = ({ children }) =>
-  <div className={styles.loader}>
-    <div className={styles.icon}>
-      <div className={styles.top}>
-        <BufferTopIcon />
-      </div>
-      <div className={styles.middle}>
-        <BufferMiddleIcon />
+const iconStyle = {
+  position: 'absolute',
+  top: 0,
+  left: 0,
+  width: '1rem',
+  height: '1rem',
+  transition: 'all 1000ms ease-in-out',
+  opacity: 1,
+};
+
+const hiddenIconStyle = {
+  ...iconStyle,
+  opacity: 0,
+};
+
+/* eslint-disable react/prop-types */
+
+const generateKeyFrames = ({ component, delay = 0 }) =>
+  <Keyframes delay={delay} loop>
+    <Frame
+      duration={1000}
+    >
+      <div
+        style={hiddenIconStyle}
+      >
+        {component}
       </div>
-      <div className={styles.bottom}>
-        <BufferBottomIcon />
+    </Frame>
+    <Frame
+      duration={1000}
+    >
+      <div
+        style={iconStyle}
+      >
+        {component}
       </div>
+    </Frame>
+  </Keyframes>;
+
+/* eslint-enable react/prop-types */
+
+const Loader = ({ children }) =>
+  <div
+    style={{
+      textAlign: 'center',
+    }}
+  >
+    <div
+      style={{
+        position: 'relative',
+        margin: '0 auto 1rem auto',
+        width: '1rem',
+        height: '1rem',
+      }}
+    >
+      {generateKeyFrames({
+        component: <BufferTopIcon />,
+      })}
+      {generateKeyFrames({
+        component: <BufferMiddleIcon />,
+        delay: 250,
+      })}
+      {generateKeyFrames({
+        component: <BufferBottomIcon />,
+        delay: 500,
+      })}
```

Comment:
Holy heck... this implementation feels a league ahead of what we had before. Love this, @hharnisc!